### ISSUE TITLE
[BUG] When the zt-attchments are not matched between Obisidian and Zotero

### ISSUE THREAD

**Issue by WayenVan** [PKM-er/obsidian-zotlit#382]
Hi, This issue happens when I have two different itemID of the same Item in different device

When I update the note, it jump out a selection manual 

**Actaul behaviour**:

![image](https://github.com/user-attachments/assets/68b53dbe-4b80-4d70-9573-3840a04e3b93)

and after I select one of them, it poped up error:

![image](https://github.com/user-attachments/assets/7ca0a834-9feb-4980-a0fa-357161245ffb)

**Expect behaviour**

At least message that I have a wrong zt-attchments ID in my notes, and popup possible IDs in current dataset

**Debug info""

![image](https://github.com/user-attachments/assets/74e8c0de-36f4-41e8-aa51-2a24235c3a60)

here when no item in `l` could match with `u` (indicate `c.length ===0`) cause this weird behavior. 

**PS** 

Does anyone know why the same Zotero item could have different itemID among different devices? is that an sync error or just means 
itemID could not be regarded as "unique identity"?

**Comment by WayenVan** [comment]
Update, the itemID is different across the different devices, so need to update everytime I switch devices, it is quite inconvenient.

**Comment by dcgk** [comment]
Hi! 
I am coming to this from a different perspective/usecase, but hope what I have figured out might be helpful to some.

I have been trying to find an efficient way to manage annotations from pdf attachments as I move them from being managed internally by Zotero as attachments to being linked pdf files in Zotero. 
The benefit of managed pdf attachments is that they are synced accross devices with Zotero Sync. For example: it is possible to make annotations in a pdf on your iPad using the Zotero app. Zotero Sync brings those annotations back to your desktop Zotero and they show up in Obsidian via ZotLit Annotation View. It is super fast and efficient! However, Zotero storage is limited and if your library is more than 6GB you need the unlimited plan for 120USD/year. The best option is to simply get that unlimited plan and support Zotero.
if that is too much, then you need to shuffle around attachments from being managed internally to being linked- for example using a Zotero plugin such as ZotMoov. However, once you move an attachmant from being internally managed by Zotero to being a linked pdf  - the pdf attachment is assigned both a new 'itemID' and 'key' by Zotero. This happens each time, so it is not the case that attachments get reassigned the same 'itemID' and 'key' when they are moved back to being linked/attached.
This 'itemID' is the 'it.attachment.itemID' which is automatically assigned by ZotLit to the YAML frontmatter property 'zt-attachmants'  of the Literature Note when it is first created. Once your pdf attachment has a different `it.attachment.itemID`, the updating of your Literature Note does not function anymore. You can manually update the YAML frontmatter property 'zt-attachmants'  to the new 'it.attachment.itemID' and then updating of your Literature Note will work again.
However, the old links from previous annotations will not work anymore. These links use the  'it.attachment.key' which is made up of eight capitalized letter or digits. It looks identical to the 'it.key' which is the item (book/paper/article) in Zotero and is used by ZotLit in the YAML frontmatter property 'zotero-key'  But the two are very different! 'it.key' identifies the item. 'it.attachmant.key' identifies the attachment. If you want your old links to work after the 'it.attachment.key' has changed you simply need to run a 'find and replace' over your markdown note. It is possible to automate this with a Templater template.

Ok, enough for now. hopefully this is helpful to someone.
Greetings!

**Comment by wangzhenchi97** [comment]
Yes, I am having a similar issue as well. I'm using JianGuoYun as an external link store instead of using webdav. zt-attachments will be different on different devices, which will cause the update to fail. It may be possible to select attachments according to the logic of the note creation when updating, which may make more sense. This avoids having to retrieve annotations based on different zt-attachments values

**Comment by aidenlx** [comment]
In v2, a note is linked to the Zotero item itself, and ZotLit looks up its current attachments straight from your library each time, so this shouldn't happen anymore. If a saved attachment choice is no longer valid, ZotLit now falls back to an available one instead of failing outright.

v2 beta is available now: https://zotlit.aidenlx.site/docs/install-zotlit
Migration guide: https://zotlit.aidenlx.site/docs/how-to/migrate-from-v1
